File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    
    int count = 0;
    
    for (const auto& i : s) {
        if (i == '(')
            count++;
        else if (i == ')')
            count--;
        
        if (count < 0)
            return false;
    }
    if (count == 0)
        return true;
    else 
        return false;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    
    stack = []
    for c in s:
        if c == '(':
            stack.append(c)
        elif c == ')':
            if len(stack) == 0:
                return False
            del stack[-1]
    if len(stack) != 0:
        return False

    return True